src/System.Text.Json/tests/JsonTestHelper.cs

```diff
@@ -212,6 +213,24 @@ public static byte[] ReaderLoop(int inpuDataLength, out int length, ref Utf8Json
             {
                 JsonTokenType tokenType = json.TokenType;
                 ReadOnlySpan<byte> valueSpan = json.HasValueSequence ? json.ValueSequence.ToArray() : json.ValueSpan;
+                if (json.HasValueSequence)
+                {
+                    Assert.True(json.ValueSpan.IsEmpty);
```

Comment:
Asserting that it's `== default` is technically a stronger claim, if you wish to make it.